identity-service/src/notifications/notificationQueue.js

```diff
@@ -62,51 +62,65 @@ async function _sendNotification (notifFn, bufferObj, logger) {
 async function drainPublishedMessages (logger) {
   logger.info(`[notificationQueue:drainPublishedMessages] Beginning processing of ${pushNotificationQueue.PUSH_NOTIFICATIONS_BUFFER.length} notifications...`)
 
+  let numProcessedNotifs = 0
   for (let bufferObj of pushNotificationQueue.PUSH_NOTIFICATIONS_BUFFER) {
     if (bufferObj.types.includes(deviceType.Mobile)) {
-      await _sendNotification(sendAwsSns, bufferObj, logger)
+      const numSentNotifs = await _sendNotification(sendAwsSns, bufferObj, logger)
+      numProcessedNotifs += numSentNotifs
     }
     if (bufferObj.types.includes(deviceType.Browser)) {
-      await Promise.all([
+      const numSentNotifsArr = await Promise.all([
```

Comment:
I don't think _sendNotifications resolves anything, I think you need to get the result from the race (and also handle the reject case)

Reply:
ah good call - totally missed the _sendNotification layer